cmake/ProjectCompilerPostConfig.cmake

```diff
@@ -14,6 +14,16 @@ IF (${Trilinos_ENABLE_Kokkos})
   set(Kokkos_ENABLE_Debug_Bounds_Check ${KOKKOS_ENABLE_DEBUG}
     CACHE BOOL
     "Enable bounds checking in Kokkos array classes.")
+  set(Kokkos_ENABLE_Profiling_DEFAULT ON)
+  if (DEFINED TPL_ENABLE_DLlib)
```

Comment:
The  `DEFINED` property is a bit fragile on reconfigures, in my experience.  In this case I would use:

```
if ( (NOT "${TPL_ENABLE_DLlib}" STREQUAL "") AND (NOT TPL_ENABLE_DLlib) )
  # Explicitly disabled DLib TPL
 ...
endif()
```

But this logic may be okay in this case.  I would have to test configuring and reconfiguring for all the test cases.

Note that the auto PR tester will **not** test reconfigures.